### PR TITLE
Fix the uncompatibility with background regestration context

### LINKED ISSUE #82: java.lang.NullPointerException when using audio_service
When running AudioService.start by flutter plugin audio_service, the app crashes. 

The following exception about FlutterDownloaderPlugin is thrown :
```
java.lang.NullPointerException: Attempt to invoke virtual method 'android.app.Application android.app.Activity.getApplication()' on a null object reference
E/CrashReport(  811): 	at vn.hunghd.flutterdownloader.FlutterDownloaderPlugin.registerWith(FlutterDownloaderPlugin.java:67)
E/CrashReport(  811): 	at io.flutter.plugins.GeneratedPluginRegistrant.registerWith(GeneratedPluginRegistrant.java:33)
E/CrashReport(  811): 	at com.letabc.zaxt.MainApplication.registerWith(MainApplication.java:18)
E/CrashReport(  811): 	at com.ryanheise.audioservice.AudioServicePlugin$ClientHandler$4.onPlay(AudioServicePlugin.java:267)
E/CrashReport(  811): 	at com.ryanheise.audioservice.AudioService$MediaSessionCallback$1.run(AudioService.java:525)
E/CrashReport(  811): 	at com.ryanheise.audioservice.AudioService$MediaSessionCallback.play(AudioService.java:541)
E/CrashReport(  811): 	at com.ryanheise.audioservice.AudioService$MediaSessionCallback.onPlay(AudioService.java:523)
E/CrashReport(  811): 	at android.support.v4.media.session.MediaSessionCompat$Callback$StubApi21.onPlay(MediaSessionCompat.java:1405)
E/CrashReport(  811): 	at android.support.v4.media.session.MediaSessionCompatApi21$CallbackProxy.onPlay(MediaSessionCompatApi21.java:196)
E/CrashReport(  811): 	at android.media.session.MediaSession$CallbackMessageHandler.handleMessage(MediaSession.java:1219)
E/CrashReport(  811): 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/CrashReport(  811): 	at android.os.Looper.loop(Looper.java:150)
E/CrashReport(  811): 	at android.app.ActivityThread.main(ActivityThread.java:5546)
E/CrashReport(  811): 	at java.lang.reflect.Method.invoke(Native Method)
E/CrashReport(  811): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:794)
E/CrashReport(  811): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:684)

```

### PR DESCRIPTION
**Pull request by Mahfoud047**
This commit fixes #82

### CODE REVIEW

**Comment by samuelg123** [comment]
Hope this fix will merged asap

**Comment by hnvn** [comment]
I'll look at it this weekend

File: android/src/main/java/vn/hunghd/flutterdownloader/FlutterDownloaderPlugin.java [hunk 1]
```diff
@@ -65,7 +65,8 @@ private FlutterDownloaderPlugin(Context context, BinaryMessenger messenger) {
     @SuppressLint("NewApi")
     public static void registerWith(PluginRegistry.Registrar registrar) {
         final FlutterDownloaderPlugin plugin = new FlutterDownloaderPlugin(registrar.context(), registrar.messenger());
-        registrar.activity().getApplication().registerActivityLifecycleCallbacks(plugin);
+        Application application = (Application) registrar.context().getApplicationContext();
```
Comment (hnvn) [inline android/src/main/java/vn/hunghd/flutterdownloader/FlutterDownloaderPlugin.java]: What if we call to `activeContext()` instead of `context()`?

Comment (Mahfoud047) [reply android/src/main/java/vn/hunghd/flutterdownloader/FlutterDownloaderPlugin.java]: Yeah I think that will be enough, but I don't remember if I tried that out or not.

Comment (Mahfoud047) [reply android/src/main/java/vn/hunghd/flutterdownloader/FlutterDownloaderPlugin.java]: In which line?

Comment (hnvn) [reply android/src/main/java/vn/hunghd/flutterdownloader/FlutterDownloaderPlugin.java]: I mean why don't use `registrar.activeContext()`?

Comment (Mahfoud047) [reply android/src/main/java/vn/hunghd/flutterdownloader/FlutterDownloaderPlugin.java]: I tried but didn't work.

Comment (hnvn) [reply android/src/main/java/vn/hunghd/flutterdownloader/FlutterDownloaderPlugin.java]: OK, Thanks for your contribution

**Pull request merged**